src/main/java/com/synopsys/integration/detect/lifecycle/run/step/SignatureScanStepRunner.java

```diff
@@ -122,5 +137,19 @@ private ScanBatchRunnerUserResult findUserProvidedScanBatchRunner(Optional<File>
         }
         return ScanBatchRunnerUserResult.none();
     }
+    
+    private void addScansToWaitFor(Set<String> scanIdsToWaitFor, SignatureScanOuputResult signatureScanOutputResult, Gson gson) throws IOException {
+        List<ScanCommandOutput> outputs = signatureScanOutputResult.getScanBatchOutput().getOutputs();
+
+        for (ScanCommandOutput output : outputs) {
+                File specificRunOutputDirectory = output.getSpecificRunOutputDirectory();
+                String scanOutputLocation = specificRunOutputDirectory.toString() + "/output/scanOutput.json";
+                Reader reader = Files.newBufferedReader(Paths.get(scanOutputLocation));
+
+                SignatureScanRapidResult result = gson.fromJson(reader, SignatureScanRapidResult.class);
+
+                scanIdsToWaitFor.add(result.scanId);
```

Comment:
What do we do when we get status completion for some scans? Should we keep polling for the remaining scans?

Reply:
this code will simply add it to a set to be polled for later. This set is later used to do the actual polling. We'll poll for the first scan in the set and wait for it to return a completed status. If it never does that we will timeout after a default...I think 3 minutes? However this scan resolves we'll then poll for the next one in the set if there is one and repeat the process.